cmd2/cmd2.py

```diff
@@ -176,10 +158,37 @@ def cat_decorator(func):
     return cat_decorator
 
 
+def _get_command_arg_list(to_parse: Union[Statement, str], preserve_quotes: bool) -> List[str]:
+    """
+    Called by the argument_list and argparse wrappers to retrieve just the arguments being
+    passed to their do_* methods as a list.
+
+    :param to_parse: what is being passed to the do_* method. It can be one of two types:
+                     1. An already parsed Statement
+                     2. An argument string in cases where a do_* method is explicitly called
+                        e.g.: Calling do_help('alias create') would cause to_parse to be 'alias create'
+
+    :param preserve_quotes: if True, then quotes will not be stripped from the arguments
+    :return: the arguments in a list
+    """
+    if isinstance(to_parse, Statement):
```

Comment:
You can add an initial if to check if it is an instance of list and if so, then just return it as-is.  Doing this will make the decorator idempotent so that you can call it twice and leave in the unit test.

Reply:
That being said, I don't really see a use case for calling this decorator twice ...